transport/kubernetes/src/main/kotlin/KubernetesSenderConfig.kt

```diff
@@ -335,9 +335,11 @@ data class KubernetesSenderConfig(
          * annotations from them. Ignore non-existing or invalid variables, but log them.
          */
         private fun createAnnotations(variableNames: String): Map<String, String> {
-            val (annotations, invalid) = variableNames.split(splitAnnotationVariablesRegex)
-                .map { variable -> variable to (System.getenv(variable).orEmpty()) }
-                .partition { KEY_VALUE_SEPARATOR in it.second }
+            val (annotations, invalid) = variableNames.takeUnless { it.isEmpty() }
```

Comment:
How about instead returning early in that case, like

     variableNames.ifEmpty { return emptyMap() }

and leaving the rest of the code as-is?